clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -111,6 +124,24 @@ public synchronized Cluster fetch() {
         return cache.cluster();
     }
 
+    /**
+     * Increment the nodesTriedSinceLastSuccessfulRefresh
+     */
+    public synchronized void incrementNodesTriedSinceLastSuccessfulRefresh() {
+        this.nodesTriedSinceLastSuccessfulRefresh++;
+    }
+
+    /**
+     * Whether the client should update the cluster metadata
+     * @param nowMs
+     * @return true if client hasn't refreshed cluster metadata for maxClusterMetadataExpireTimeMs and
+     * has tried connecting to at least half of the nodes in current node set
+     */
+    public synchronized boolean shouldUpdateClusterMetadata(long nowMs) {
```

Comment:
rename to shouldUpdateClusterMetadataFromBoostrap or similiary?

Reply:
sure.